### PR TITLE
Adding radius protocol selection for user manager servers using radius

### PR DESCRIPTION
**Pull request by ubawurinna**
Adding radius protocol selection to system/user manager/servers
When selecting 'Radius' as server type, the user can now choose radius protocol instead of stuck on PAP.
Supported protocol are PAP, MD5-CHAP, MS-CHAPv1, MS-CHAPv2.

### CODE REVIEW

**Comment by rbgarga** [comment]
It cannot be merged automatically, can you please rebase your fork?

**Comment by ubawurinna** [comment]
I've rebase my fork

**Comment by doktornotor** [comment]
This got yet agian fscked up by bootstrap. 

@ubawurinna - You want to rebase this yet again (not even sure whether possible)? If you are tired of this BS, please tell me and I'll try to resubmit this PR manually.

**Comment by ubawurinna** [comment]
I've rebase my fork

File: src/usr/local/openssl/openssl.cnf [hunk 1]
```diff
@@ -1 +0,0 @@
-../../../etc/ssl/openssl.cnf
```
Comment (rbgarga) [inline src/usr/local/openssl/openssl.cnf]: Looks like it was accidentally changed, please revert

Comment (rbgarga) [inline src/usr/local/openssl/openssl.cnf]: Looks like it was accidentally changed, please revert

File: src/etc/inc/auth.inc [hunk 2]
```diff
@@ -59,6 +59,7 @@
  * NOTE : Portions of the mschapv2 support was based on the BSD licensed CHAP.php
  * file courtesy of Michael Retterklieber.
  */
+require_once("CHAP.inc");
```
Comment (rbgarga) [inline src/etc/inc/auth.inc]: Please move this require inside the function where it's needed

File: src/etc/inc/auth.inc [hunk 3]
```diff
@@ -1335,29 +1336,64 @@ function ldap_backed($username, $passwd, $authcfg) {
...
 		$radiusservers[0]['sharedsecret'] = $authcfg['radius_secret'];
 		$radiusservers[0]['timeout'] = $authcfg['radius_timeout'];
+
+		$radius_protocol = $authcfg['radius_protocol'];
 	} else {
 		return false;
 	}
 
+    // Create our instance
+    $classname = 'Auth_RADIUS_' . $radius_protocol;
```
Comment (rbgarga) [inline src/etc/inc/auth.inc]: Please fix style, indent with one tab

File: src/etc/inc/auth.inc [hunk 4]
```diff
@@ -1335,29 +1336,64 @@ function ldap_backed($username, $passwd, $authcfg) {
...
+    $rauth = new $classname($username, $password);
+
 	/* Add new servers to our instance */
 	foreach ($radiusservers as $radsrv) {
 		$timeout = (is_numeric($radsrv['timeout'])) ? $radsrv['timeout'] : 5;
 		$rauth->addServer($radsrv['ipaddr'], $radsrv['port'], $radsrv['sharedsecret'], $timeout);
 	}
 
+	// Construct data package
+    $rauth->username = $username;
```
Comment (rbgarga) [inline src/etc/inc/auth.inc]: Please fix indent

File: src/etc/inc/auth.inc [hunk 5]
```diff
@@ -1335,29 +1336,64 @@ function ldap_backed($username, $passwd, $authcfg) {
...
+            $rauth->challenge = $crpt->authChallenge;
+            $rauth->peerChallenge = $crpt->peerChallenge;
+            $rauth->chapid = $crpt->chapid;
+            $rauth->response = $crpt->challengeResponse();
+            break;
+
+        default:
+            $rauth->password = $password;
+            break;
+    }
```
Comment (rbgarga) [inline src/etc/inc/auth.inc]: Please fix indent

**Comment by rbgarga** [comment]
Please write an upgrade code to src/etc/inc/upgrade_config.inc and bump config version in src/etc/inc/globals.inc to setup old configs as "PAP" to make sure it will keep working as it was before

**Comment by ubawurinna** [comment]
I have made the change.
Could you please remove/ignore the etc/ssl/openssl.cnf change, I can not do it myself !

**Pull request closed without merging**